### ISSUE TITLE
Git LFS v1.2

### ISSUE THREAD

**Issue by technoweenie** [github/git-lfs#844]
Here are the fixes slated for v1.2:

* [x] netrc support https://github.com/github/git-lfs/pull/715 (@rubyist)
* [x] `clone` command #988 (@sinbad)
* [x] support self-signed certs #1067 (@sinbad)
* [x] support sslverify option for specific hosts #1081 (@sinbad)
* [x] fix clean/smudge filter string for files starting with `-` #1083 (@epriestley)
* [x] fix silent failure to push LFS objects when ref matches a filename in the working copy #1096 (@epriestley)
* [x] fix problems with using LFS in symlinked folders #818 (@sinbad)
* [x] touch existing git tracked files when tracked in LFS so they are flagged as modified #1104 (@sinbad)
* [x] support for git references #1007 (@jlehtnie)
* [x] fix git lfs push silently misbehaving on ambiguous refs; fail like git push instead #1118 (@sinbad)
* [x] whitelist `lfs.*.access` config in local `~/.lfsconfig` #1122 (@rjbell4)
* [x] Only write the encoded pointer information to Stdout #1105 (@sschuberth)
* [x] Use hardcoded auth from remote or lfs config when accessing the storage api #1136 (@technoweenie, @jonmagic)
* [x] SSH should be called more strictly with command as one argument #1134 (@sinbad)
* [x] Stop transferring duplicate objects on major push or fetch operations on multiple refs. #1128 (@technoweenie)
* [x] Support `url.*.insteadof` git config options #1117 (@artagnon)
* [ ] zsh auto completion https://github.com/github/git-lfs/pull/1133

**Comment by sinbad** [comment]
I'm thinking maybe it would be worth trying to bring the significant speed improvements from 'git lfs clone' to the other main workflow tasks that trigger downloads, like pulling or checking out a different branch. Unfortunately we already used 'git lfs pull' and 'git lfs checkout', maybe we either create new wrapper commands like 'git lfs gitpull' or use an option to the existing commands to turn it into a wrapper command e.g. 'git lfs pull --git'. Any thoughts?

**Comment by bazooka241** [comment]
You will excuse me busy place.
The other day I will try to was sorry I have to unpleasant feelings in my selfish behavior, be a twice like this does not occur.
Did not Thank really sorry.

**Comment by larsxschneider** [comment]
@sinbad I agree that would make sense. However, I think "git lfs checkout" and "git lfs pull" should be triggering the corresponding git operations by default. Although this change would then brake backwards compatibility (how big of a problem would that be in this particular case?). How about a git-lfs config flag that defines if these commands trigger their respective git operations as well? The flag could be off by default to keep the current behavior.

**Comment by sinbad** [comment]
@larsxschneider it's not such a big deal with `git lfs checkout` because it already matches `git checkout` behaviour for file paths, it just doesn't support the additional options for switching or creating branches yet. Although paths might need to be after a `--` marker for disambiguation like `git checkout` which is a breaking change.

For `git lfs pull` it's quite a big change since until then people won't have expected this to change the non-LFS files in their working copy. Also we still need a command which just does the lfs fetch & checkout, but perhaps you're right that this should be the one to change name  or require an additional option. For consistency it would probably also mean `git lfs fetch` would have to change to fetch commits first too (principle of least surprise if everything else that sounds like a git command does that).

I've written up my proposal in #1059

**Comment by sinbad** [comment]
Another idea for 1.2: maybe we should drop the non-batch route now? There are edge cases which can cause the 'sticky' `batch=false` setting to be enabled which can break usage in non-intuitive ways, such as #993. Since no recent server implementations should be reliant on the non-batch route I think we should eliminate it.

**Comment by technoweenie** [comment]
We still have users hitting those endpoints, unfortunately :(

**Comment by sinbad** [comment]
You mean clients hitting them on GH, or other server implementations only supporting non-batch? If it's clients then updating the client will phase those out over time, they're probably only using them because of the 'sticky' `batch=false` setting getting set one time but don't realise, and really they don't need it anymore? I only meant dropping it on the client.

**Comment by technoweenie** [comment]
I'd like to push out a release next week. What is nearly ready, and just needs some review?

0. Add ref information to upload request #969 - I think this is nearly ready. 
0. zsh auto completion #727 - This _looks_ ready. I use ZSH, so I'll try it out.
0. touch existing git tracked files when tracked in LFS so they are flagged as modified #1104 (@sinbad)
0. support for git references #1007 (@jlehtnie)
0. Accept raw remote URLs as valid #1085 (@epriestley)
0. fix git lfs push silently misbehaving on ambiguous refs; fail like git push instead #1100

These would be nice, but could wait for a v1.3 release. We could include them if a champion rises up. I personally won't touch them until the above issues are done.

0. authenticated property #960
0. add all lfs.* git config keys to git lfs env output

So, what can we chop off from the above list? These are things that haven't had any work done. Some are pretty vague, or may be irrelevant now.

0. Everything under "Refactoring" and "Investigate"
0. better support for network issues #848
0. credential output hidden while transferring files #847
0. Better SSH remote handling #295 #278
0. Git LFS is unaware of url configurations #864
0. Fails to checkout when Git repo is used via symlink #866
0. Investigate pre-push error handling #1066

@sinbad: If you agree, I'll update the original issue body at the top.

Anyone else: feel free to chime in :)

**Comment by sinbad** [comment]
Looks good to me, although I think #1085 is pretty niche; if the URL doesn't match any remote it can really only do a brute-force push of everything. 

I don't use `zsh` but I can look at #1007 a bit more.

**Comment by technoweenie** [comment]
> Looks good to me, although I think #1085 is pretty niche; if the URL doesn't match any remote it can really only do a brute-force push of everything.

I think that's fine. Seems better than not doing anything, at least.

> I don't use zsh but I can look at #1007 a bit more.

I do use zsh, but I'd appreciate the help :)

I have no qualms holding these, or other issues if any problems come up. I've updated the issue body at the top with the CURRENT list, and will eventually find a place on the roadmap (after v1.2 ships) for the chopped items.

**Comment by epriestley** [comment]
I'm planning to return to #1085 when I have a chance, but probably won't get there in the next few days.

(Our downstream LFS server implementation is available as a prototype and waiting on "does this kinda work?" user feedback, so I probably won't swing back around to LFS until we get that and I can knock out some other adjacent work at the same time.)

**Comment by technoweenie** [comment]
I moved #1085 to the "nice to have" list. It had a bit more work to do than I remembered.

**Comment by technoweenie** [comment]
Shipped! https://github.com/github/git-lfs/releases/tag/v1.2.0

/cc @github/desktop
